### PR TITLE
test: Add retries to get firecracker PID in new ns

### PR DESCRIPTION
**Pull request by zulinx86**
## Changes

- Adds reties to `pid_in_new_ns`.

## Reason

When `pid_in_new_ns` is referenced before the pid file is created by jailer with `--new-pid-ns` flag, it returns `None` and some jailer integration tests fail.
```
___________________ test_new_pid_ns_resource_limits[ubuntu] ____________________
--
  | integration_tests/security/test_jail.py:629: in test_new_pid_ns_resource_limits
  | check_limits(fc_pid, NOFILE, FSIZE)
  | fc_pid     = None
  | test_microvm = <framework.microvm.Microvm object at 0x7fd92e890d30>
  | test_microvm_with_api = <framework.microvm.Microvm object at 0x7fd92e890d30>
  | integration_tests/security/test_jail.py:348: in check_limits
  | (soft, hard) = resource.prlimit(pid, resource.RLIMIT_NOFILE)
  | E   TypeError: 'NoneType' object cannot be interpreted as an integer
  | fsize      = 2097151
  | no_file    = 1024
  | pid        = None
```

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- ~~[ ] If a specific issue led to this PR, this PR closes the issue.~~
- [x] The description of changes is clear and encompassing.
- ~~[ ] Any required documentation changes (code and docs) are included in this PR.~~
- ~~[ ] API changes follow the [Runbook for Firecracker API changes][2].~~
- ~~[ ] User-facing changes are mentioned in `CHANGELOG.md`.~~
- [x] All added/changed functionality is tested.
- ~~[ ] New `TODO`s link to an issue.~~
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by roypat — DISMISSED** [review]

File: tests/framework/microvm.py [hunk 1]
```diff
@@ -339,20 +337,18 @@ def started(self):
...
-        fc_pid = None
-
-        pid_file_path = f"{self.jailer.chroot_path()}/{FC_PID_FILE_NAME}"
-        if os.path.exists(pid_file_path):
-            # Read the PID stored inside the file.
-            with open(pid_file_path, encoding="utf-8") as file:
-                fc_pid = int(file.readline())
+        # Check if the pid file exists.
+        pid_file_path = Path(f"{self.jailer.chroot_path()}/{FC_PID_FILE_NAME}")
+        assert os.path.exists(pid_file_path)
```
Comment (pb8o) [inline tests/framework/microvm.py:349]: can also do `pid_file_path.exists()`

File: tests/framework/microvm.py [hunk 2]
```diff
@@ -339,20 +337,18 @@ def started(self):
...
 
-        return fc_pid
+        # Read the PID stored inside the file.
+        return int(pid_file_path.read_text(encoding="utf-8"))
```
Comment (pb8o) [inline tests/framework/microvm.py:352]: super-nit" `ascii` would be enough

**Review by pb8o — DISMISSED** [review]

File: tests/framework/microvm.py (hunk 1 above)
Comment (zulinx86) [reply tests/framework/microvm.py:349]: I did know this, thanks!

File: tests/framework/microvm.py (hunk 2 above)
Comment (zulinx86) [reply tests/framework/microvm.py:352]: indeed. will fix.

**Review by roypat — APPROVED** [review]

**Review by pb8o — APPROVED** [review]

**Pull request merged**